packages/core/src/amazonqGumby/chat/controller/controller.ts

```diff
@@ -269,29 +269,41 @@ export class GumbyController {
     }
 
     private async handleLanguageUpgrade(message: any) {
-        try {
-            await this.beginTransformation(message)
-            const validProjects = await this.validateLanguageUpgradeProjects(message)
-            if (validProjects.length > 0) {
-                this.sessionStorage.getSession().updateCandidateProjects(validProjects)
-                await this.messenger.sendLanguageUpgradeProjectPrompt(validProjects, message.tabID)
+        await telemetry.codeTransform_submitSelection.run(async () => {
+            telemetry.record({
+                codeTransformSessionId: CodeTransformTelemetryState.instance.getSessionId(),
```

Comment:
Looks like session ID is mostly if not always required in our metrics; not sure why because I don't use it much. I'll fix this in the commons package and then submit a follow up PR here along with some other telemetry-related improvements shortly.